### PR TITLE
BD-2366: Data transformation GA - MERGE 9/5

### PR DESCRIPTION
**Pull request by lydia-xie**
# Pull Request/Issue Resolution

Closes #**BD-2366**

#### Is this change associated with a Braze feature/product release?
- [X] Yes (**September 5, 2023**)
- [ ] No

---

### CODE REVIEW

File: _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+nav_title: Creating a Transformation
+article_title: Creating a Transformation
+page_order: 1
+page_type: reference
+description: "This reference article provides steps on how to create a transformation using Braze Data Transformation."
+---
+
+# Creating a transformation
+
+> Braze Data Transformation allows you to build and manage webhook integrations to automate data flow from external platforms into Braze user profiles. This integrated user data can then power even more sophisticated marketing use cases.
```
Comment (lydia-xie) [inline _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md:11]:
```suggestion
> Braze Data Transformation enables you to build and manage webhook integrations to automate data flow from external platforms into Braze user profiles. This integrated user data can then power even more sophisticated marketing use cases.
```

File: _docs/_user_guide/data_and_analytics/data_transformation/overview.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+Many modern-day platforms have “webhooks,” or real-time API notifications, to send information about a new event or new data from one platform to another. Data Transformation provides:
+
+- A Braze address to receive such webhooks.
+- Capabilities to transform the webhook’s payload with JavaScript code, so you can create a valid Braze `/users/track` call. You can choose what information to use from the webhook, and how you want the data represented on Braze user profiles as user attributes, events, or purchases.
+- Capabilities to send the compiled `/users/track` call.
+
+The end result is a webhook integration that connects a source platform of your choice, by turning their webhooks into Braze user profile updates.
+
+{% details More on webhooks %}
+Webhooks are real-time notifications sent via an HTTP POST request to a specific destination. Webhooks are often used to send data from one point to another, in which the webhook can pass through data about an action that has occurred and who is involved in that action.
```
Comment (lydia-xie) [inline _docs/_user_guide/data_and_analytics/data_transformation/overview.md:26]:
```suggestion
Webhooks are real-time notifications sent via an HTTP POST request to a specific destination. Webhooks are often used to send data from one point to another, in which the webhook can pass through data about an action that has occurred and who was involved in that action.
```

File: _docs/_user_guide/data_and_analytics/data_transformation/overview.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+{% alert important %}
+To request an upgrade to Data Transformation Pro, contact your Braze account manager or select the **Request Upgrade** button in the Braze dashboard.
+{% endalert %}
+
+### Rate limits
+
+The rate limit for Braze Data Transformation is 1,000 incoming requests per minute. If you have Data Transformation Pro and need a higher rate limit, contact your Braze account manager.
+
+## Frequently asked questions
```
Comment (bre-fitzgerald) [inline _docs/_user_guide/data_and_analytics/data_transformation/overview.md:50]: Can we pull these FAQs out of the details dropdowns? We typically use headers for these now afaik

File: _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+# Creating a transformation
+
+> Braze Data Transformation enables you to build and manage webhook integrations to automate data flow from external platforms into Braze user profiles. This integrated user data can then power even more sophisticated marketing use cases.
+
+## Prerequisites 
+
+| Requirement | Description |
+| --- | --- |
+| 2FA or SSO | You must have 2FA or SSO enabled for your account. |
```
Comment (bre-fitzgerald) [inline _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md:17]:
```suggestion
| Two-factor authentication or SSO | You must have [two-factor authentication]({{site.baseurl}}/user_guide/administrative/app_settings/company_settings/security_settings/#two-factor-authentication) (2FA) or [single sign-on]({{site.baseurl}}/user_guide/administrative/app_settings/company_settings/security_settings/#single-sign-on-sso-authentication) (SSO) enabled for your account. |
```

File: _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+          "properties": {
+            "property_1": payload.purchase_1.property_1
+          }
+        }
+      ]
+    };
+
+    // After the/users/track request is assigned to brazecall, you will want to explicitly return brazecall to create an output
+    return brazecall;
+    ```
```
Comment (bre-fitzgerald) [inline _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md:118]: Adding blank space after code block

```suggestion
    ```

```

File: _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+3. Each attribute, event, and purchase object requires a user identifier, either an `external_id`, `user_alias`, email, or `braze_id`. Find the user identifier in the incoming webhook’s payload and template in that value in your transformation code via a payload line. Use dot notation to access payload object properties.<br>Template in the email address from the incoming webhook and use `email` as your identifier in the transformation code. Our example transformation code page uses this `/users/track` functionality.<br><br>
+4. Find the webhook values you’d like to represent as attributes, events, or purchases, and template those values in your transformation code via a payload line. Use dot notation to access payload object properties.<br><br>
+5. For each attribute, event, and purchase object, examine the `_update_existing_only` value. Set this to `false` if you want the transformation to create a new user that may not exist. Leave this as `true` to only update existing profiles.<br><br>
+6. Click **Validate** to return a preview of your code’s output and to check if it is an acceptable `/users/track` request.
```
Comment (bre-fitzgerald) [inline _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md:123]: Formatting consistency

```suggestion
6. Click **Validate** to return a preview of your code’s output and to check if it is an acceptable `/users/track` request.<br><br>
```

File: _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+7. Activate your transformation. For additional help with your code before activating it, contact your Braze account manager.<br><br>
+7. Have your source platform begin sending webhooks. Your transformation code will run for each incoming webhook, and user profiles will begin updating. 
+
+Your webhook integration is now complete!
+
+{% alert important %}
+Accepting email as an identifier is possible as Braze Data Transformation early access users will also be granted early access to this new `/users/track` feature to update a user profile by email address.<br><br>early access users of Braze Data Transformation who started before April 2023 may be familiar with a `get_user_by_email` function that helped with this use case. That function is deprecated.
```
Comment (bre-fitzgerald) [inline _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md:130]:
```suggestion
Accepting email as an identifier is possible as Braze Data Transformation early access users will also be granted early access to this new `/users/track` feature to update a user profile by email address.<br><br>Early access users of Braze Data Transformation who started before April 2023 may be familiar with a `get_user_by_email` function that helped with this use case. That function is deprecated.
```

File: _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+{% endalert %}
+
+{% endtab %}
+{% tab Advanced %}
+
+In this step, you will transform the webhook payload from the source platform to a JavaScript object return value. This return value must adhere to Braze’s `/users/track` request body format:
+
+- Transformation code is accepted in the JavaScript programming language. Any standard JavaScript control flow, such as if/else logic is supported.
```
Comment (bre-fitzgerald) [inline _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md:138]:
```suggestion
- Transformation code is accepted in the JavaScript programming language. Any standard JavaScript control flow, such as if/else logic, is supported.
```

File: _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+- Transformation code accesses the webhook request body via the `payload` variable. This variable is an object populated by parsing the request body JSON.
+- Any feature supported in our `/users/track` endpoint is supported, including:
+  - User attributes objects, event objects, and purchase objects
+  - Nested attributes and nested custom event properties
+  - Subscription group updates
+  - Email address as an identifier
+- Click **Validate** to return a preview of your code’s output and to check if it's an acceptable `/users/track` request.
```
Comment (bre-fitzgerald) [inline _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md:145]: This doesn't seem to apply to the same level of detail as the rest of the bullets. Suggest removing from the list and just add as a sentence after.
```suggestion

Click **Validate** to return a preview of your code’s output and to check if it's an acceptable `/users/track` request.
```

File: _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+{% endtab %}
+{% endtabs %}
+
+## Step 5: Monitor your transformation
+
+After activating your transformation, refer to the analytics on the **Transformations** page to monitor its performance.
+
+**Incoming Requests:** This is the number of webhooks received at this transformation’s URL. If incoming requests are 0, your source platform hasn’t sent over any webhooks, or the connection cannot be made.
+
+**Deliveries:** After receiving incoming requests, Data Transformation applies your transformation code to create a Braze `/users/track` request.
```
Comment (bre-fitzgerald) [inline _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md:159]: Bullets here for readability? 🤔 

```suggestion
- **Incoming Requests:** This is the number of webhooks received at this transformation’s URL. If incoming requests are 0, your source platform hasn’t sent over any webhooks, or the connection cannot be made.
- **Deliveries:** After receiving incoming requests, Data Transformation applies your transformation code to create a Braze `/users/track` request.
```

File: _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+
+The number of deliveries will never be greater than the number of incoming requests. However, it is a good goal to have 100% of incoming requests leading to deliveries.
+
+- If deliveries are 0, check your transformation code to ensure there are no syntax errors and that it compiles. Then, check whether the output is a valid `/users/track` request.
```
Comment (bre-fitzgerald) [inline _docs/_user_guide/data_and_analytics/data_transformation/creating_a_transformation.md:163]: Perhaps add a troubleshooting header before this bulleted list? Up to you

```suggestion
### Troubleshooting

- If deliveries are 0, check your transformation code to ensure there are no syntax errors and that it compiles. Then, check whether the output is a valid `/users/track` request.
```

**Review by bre-fitzgerald — CHANGES_REQUESTED** [review]
@lydia-xie looks good! Just left a few minor suggestions and comments for your review.

File: _docs/_user_guide/data_and_analytics/data_transformation/overview.md (hunk 3 above)
Comment (lydia-xie) [reply _docs/_user_guide/data_and_analytics/data_transformation/overview.md:50]: Done!

**Review by bre-fitzgerald — APPROVED** [review]

**Pull request merged**